### PR TITLE
Add Zig wrapper for kernel32.DeviceIoControl

### LINKED ISSUE #5944: NtFsControlFile vs NtDeviceIoControlFile
@daurnimator
> When would you want to use `NtFsControlFile` vs `NtDeviceIoControlFile`?
> I've always used the latter (but without a good reason)

@kubkon
> That's an excellent question, thanks @daurnimator! Originally I wanted to use `NtDeviceIoControlFile` but trial-and-error, and the docs revealed that for reparse points processing we should use `NtFsControlFile` which is similar to `NtDeviceIoControlFile` but specifically targets filesystem drivers. Also according to [docs](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/nf-ntifs-ntfscontrolfile#remarks), we should use it to read/set/update reparse points, and any IO control code starting with `FSCTL_xxx`.

_From https://github.com/ziglang/zig/pull/5937#issuecomment-664891363_

### PR DESCRIPTION
**Pull request by kubkon**
This commit adds a Zig wrapper for `kernel32.DeviceIoControl` which
applies ReactOS logic for deciding whether to use
`ntdll.NtDeviceIoControlFile` or `ntdll.NtFsControlFile` based on the
value of passed `IO_CONTROL_CODE`. The decision logic is based on the
logic found in ReactOS found in the following [link].

Thanks to @daurnimator  for finding this bit in ReactOS!

[link]: https://doxygen.reactos.org/d3/d74/deviceio_8c.html

Closes #5944

### CODE REVIEW

File: lib/std/os/windows.zig [hunk 1]
```diff
@@ -215,24 +215,28 @@ pub fn CreateEventExW(attributes: ?*SECURITY_ATTRIBUTES, nameW: [*:0]const u16,
...
+pub fn DeviceIoControl(
     h: HANDLE,
-    fsControlCode: ULONG,
+    ioControlCode: ULONG,
     in: ?[]const u8,
     out: ?[]u8,
-) FsControlFileError!void {
+) DeviceIoControlError!void {
+    // Logic from: https://doxygen.reactos.org/d3/d74/deviceio_8c.html
+    const syscall = if ((ioControlCode >> 16) == FILE_DEVICE_FILE_SYSTEM) ntdll.NtFsControlFile else ntdll.NtDeviceIoControlFile;
```
Comment (daurnimator) [inline lib/std/os/windows.zig:231]: does this end up using a function pointer instead of a normal call?

Comment (andrewrk) [reply lib/std/os/windows.zig:231]: yes

Comment (daurnimator) [reply lib/std/os/windows.zig:231]: Would it be better to have two function call code paths then?

Comment (andrewrk) [reply lib/std/os/windows.zig:231]: yes

Comment (kubkon) [reply lib/std/os/windows.zig:231]: That's a good point. I'll provide a quickfix ASAP before we forget to apply this!

**Pull request merged**